source/extensions/clusters/redis/redis_cluster.cc

```diff
@@ -77,17 +79,21 @@ void RedisCluster::updateAllHosts(const Upstream::HostVector& hosts_added,
 
 void RedisCluster::onClusterSlotUpdate(const std::vector<ClusterSlot>& slots) {
   Upstream::HostVector new_hosts;
-  SlotArray slots_;
 
   for (const ClusterSlot& slot : slots) {
-    new_hosts.emplace_back(new RedisHost(info(), "", slot.master_, *this, true));
+    new_hosts.emplace_back(new RedisHost(info(), "", slot.master(), *this, true));
   }
 
   std::unordered_map<std::string, Upstream::HostSharedPtr> updated_hosts;
   Upstream::HostVector hosts_added;
   Upstream::HostVector hosts_removed;
-  if (updateDynamicHostList(new_hosts, hosts_, hosts_added, hosts_removed, updated_hosts,
-                            all_hosts_)) {
+  bool host_updated = updateDynamicHostList(new_hosts, hosts_, hosts_added, hosts_removed,
+                                            updated_hosts, all_hosts_);
+  bool slot_updated = lb_factory_ ? lb_factory_->onClusterSlotUpdate(slots, updated_hosts) : false;
```

Comment:
When is lb_factory_ null? You pass as part of the constructor?

Reply:
This is due to the back compat issue, this can be removed after the back compat is used.